pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook_test.go

```diff
@@ -83,6 +85,88 @@ func TestSyncSetValidate(t *testing.T) {
 			syncSet:         testSyncSet(),
 			expectedAllowed: true,
 		},
+		{
+			name:            "Test valid SecretReference kind create",
+			operation:       admissionv1beta1.Create,
+			syncSet:         testSecretReferenceSyncSet(),
+			expectedAllowed: true,
+		},
+		{
+			name:            "Test valid SecretReference kind update",
+			operation:       admissionv1beta1.Update,
+			syncSet:         testSecretReferenceSyncSet(),
+			expectedAllowed: true,
+		},
+		{
+			name:      "Test invalid SecretReference kind create",
+			operation: admissionv1beta1.Create,
+			syncSet: func() *hivev1.SyncSet {
+				ss := testSecretReferenceSyncSet()
+				ss.Spec.SecretReferences[0].Source.Kind = "wrongkind"
```

Comment:
Separate this into two distinct sub tests so that we verify that we are checking both `Source` and `Target` and not just one of the two.